akvo/rsr/tests/rest/test_indicator_period_data.py

```diff
@@ -1669,3 +1670,280 @@ def test_query_audit_trail(self):
+            .with_partner(org, Partnership.IATI_REPORTING_ORGANISATION)\
+            .with_results([{
+                'title': 'Result  #1',
+                'indicators': [{
+                    'title': 'Indicator #1',
+                    'periods': [{
+                        'period_start': date(2010, 1, 1),
+                        'period_end': date(2010, 12, 31),
+                    }]
+                }]
+            }]).build()
+        period = project.get_period(period_start=date(2010, 1, 1))
+        period.add_update(user2, value=1, status='D')
+        # When
+        self.c.login(username=user1.email, password='password')
+        response = self.c.get(f'/rest/v1/project/{project.object.id}/results_framework/?format=json')
+        # Then
+        updates = response.data['results'][0]['indicators'][0]['periods'][0]['updates']
+        self.assertEqual(0, len(updates))
+
+    def test_get_results_with_collab(self):
+        # Given
+        org = self.create_organisation('Acme')
+        user1, user2 = (self.create_user_org(f'{u}@acme.org', org, 'Enumerators') for u in ['user1', 'user2'])
+        project = ProjectFixtureBuilder()\
+            .with_partner(org, Partnership.IATI_REPORTING_ORGANISATION)\
+            .with_results([{
+                'title': 'Result  #1',
+                'indicators': [{
+                    'title': 'Indicator #1',
+                    'periods': [{
+                        'period_start': date(2010, 1, 1),
+                        'period_end': date(2010, 12, 31),
+                    }]
+                }]
+            }]).build()
+        period = project.get_period(period_start=date(2010, 1, 1))
+        update = period.add_update(user2, value=1, status='D')
+        # When
+        self.c.login(username=user1.email, password='password')
+        with override_settings(NUFFIC_ROOT_PROJECT=project.object.id):
+            response = self.c.get(f'/rest/v1/project/{project.object.id}/results_framework/?format=json')
+        # Then
+        updates = response.data['results'][0]['indicators'][0]['periods'][0]['updates']
+        self.assertEqual(1, len(updates))
+        self.assertEqual(update.id, updates[0]['id'])
+
+    def test_list_updates_no_collab(self):
+        # Given
+        org = self.create_organisation('Acme')
+        user1, user2 = (self.create_user_org(f'{u}@acme.org', org, 'Enumerators') for u in ['user1', 'user2'])
+        project = ProjectFixtureBuilder()\
+            .with_partner(org, Partnership.IATI_REPORTING_ORGANISATION)\
+            .with_results([{
+                'title': 'Result  #1',
+                'indicators': [{
+                    'title': 'Indicator #1',
+                    'periods': [{
+                        'period_start': date(2010, 1, 1),
+                        'period_end': date(2010, 12, 31),
+                    }]
+                }]
+            }]).build()
+        period = project.get_period(period_start=date(2010, 1, 1))
```

Comment:
It looks like these belong in `setUp()`. They seem to be in every test.

Reply:
It's personal taste, but I don't really like using the setup method to make the test more "DRY". In a large test class, it often leaves me scrolling up and down just to understand the context of a test case.

I prefer the "DAMP" approach where I can see everything that matters to the test case in one place.

But I've made some changes in the test code to make it more "DRY"

Reply:
OK. We should probably agree on something later to make it more consistent in future code.